stuffer/s2n_stuffer.c

```diff
@@ -23,10 +23,22 @@
 #include "utils/s2n_blob.h"
 #include "utils/s2n_mem.h"
 
+bool s2n_stuffer_is_valid(const struct s2n_stuffer* stuffer)
+{
+  return S2N_OBJECT_PTR_IS_READABLE(stuffer) && 
+    s2n_blob_is_valid(&stuffer->blob) &&
+    /* <= is valid because we can have a fully written/read stuffer */
+    stuffer->read_cursor <= stuffer->blob.size &&
+    stuffer->write_cursor <= stuffer->blob.size &&
+    stuffer->read_cursor <= stuffer->write_cursor;
+    /*any combination of wiped, alloced, growable, and tainted are valid */
```

Comment:
This comment is slightly mysterious to me, I don't see how it relates to the above predicate. But perhaps it's meaningful to s2n developers.

Reply:
They are the boolean fields in the stuffer, which I don't check since they're valid either way.

Reply:
Clarified in the comment